libs/analytics/src/widget/orderLifecycleAnalytics.ts

```diff
@@ -10,56 +11,101 @@ import {
 } from '@cowprotocol/events'
 
 import { getCowAnalytics } from '../utils'
+// Local alias to keep call sites tidy in this file
+const formatUnits = formatUnitsSafe
 
 // Specialized helper function for string properties to ensure we always return a string
 const safeGetString = <T, K extends keyof T>(obj: T | undefined, key: K, fallback: string = ''): string => {
   const value = obj && obj[key] !== undefined ? obj[key] : fallback
   return String(value)
 }
 
-// TODO: Add proper return type annotation
-// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
-const getOrderPayload = (payload: BaseOrderPayload) => ({
-  walletAddress: safeGetString(payload.order, 'owner'),
-  orderId: safeGetString(payload.order, 'uid'),
-  chainId: payload.chainId.toString(),
-  sellToken: safeGetString(payload.order, 'sellToken'),
-  buyToken: safeGetString(payload.order, 'buyToken'),
-  sellAmount: safeGetString(payload.order, 'sellAmount'),
-  buyAmount: safeGetString(payload.order, 'buyAmount'),
-})
+type AnalyticsPayload = Record<string, unknown>
+
+type TokenLike = { symbol?: string; decimals?: number; address?: string }
```

Comment:
``import { TokenInfo } from '@cowprotocol/cow-sdk' can this type be used instead?